### ISSUE TITLE
Rename `tower-batch` crate

### ISSUE THREAD

**Issue by mpguerra** [ZcashFoundation/zebra#6202]
## Motivation

Once we are ready to publish a stable release for zebra we will want to publish our crates on crates.io. However, another `tower-batch` crate already exists there. 

### Specifications

We will need to rename our own `tower-batch` crate, perhaps to `tower-batch-crypto`? we can discuss on this issue. 

We will need to rename it in the docs, and replace all the links to it with the new name.

## Related Work

#6149

**Comment by teor2345** [comment]
> We will need to rename our own `tower-batch` crate, perhaps to `tower-batch-crypto`? we can discuss on this issue.

The [other `tower-batch`](https://crates.io/crates/tower-batch) is about disk write batching, but our batches run CPU tasks in parallel.

So maybe something like:
* tower-cpu-batch
* tower-parallel-batch
* tower-parallel-cpu-batch

It's not specifically tied to cryptography or cryptocurrency.

**Comment by arya2** [comment]
`zebra-tower-batch` to match the rest of the crate names in the workspace?

> The [other tower-batch](https://crates.io/crates/tower-batch) is about disk write batching, but our batches run CPU tasks in parallel.

I think disk write batching is the implied motivation but it looks similar to Zebra's `tower-batch` (see [BatchControl](https://docs.rs/tower-batch/0.1.0/tower_batch/enum.BatchControl.html) and [Batch](https://docs.rs/tower-batch/0.1.0/tower_batch/struct.Batch.html) in the other crate).

**Comment by teor2345** [comment]
> `zebra-tower-batch` to match the rest of the crate names in the workspace?

There's nothing Zebra-specific about this crate, it is tower middleware that can be used by any project.

**Comment by teor2345** [comment]
We will need to rename tower-batch in the docs, and replace all the links to it with the new name. I added this to the ticket.

**Comment by mpguerra** [comment]
I see we've published the placeholders for `tower-batch-cpu` and `tower-batch-control` to crates.io 🎉 

Now that that's done, the actual rename is lower priority until we publish a stable release.

**Comment by teor2345** [comment]
Some renames were missed in the PR, the remaining work is in #6914
